src/main/java/com/googleinterns/zoomtube/servlets/TranscriptServlet.java

```diff
@@ -37,42 +53,132 @@
 public class TranscriptServlet extends HttpServlet {
   private static final String TRANSCRIPT_XML_URL_TEMPLATE =
       "http://video.google.com/timedtext?lang=en&v=";
-  private static final String START_ATTRIBUTE = "start";
-  private static final String DURATION_ATTRIBUTE = "dur";
-  private static final String TEXT_TAG = "text";
-  private static final String TEST_VIDEO_ID = "3ymwOvzhwHs";
+  public static final String ATTR_START = "start";
+  public static final String ATTR_DURATION = "dur";
+  public static final String TAG_TEXT = "text";
+  public static final String PARAM_LECTURE = "lecture";
+  public static final String PARAM_LECTURE_ID = "id";
+  public static final String PARAM_VIDEO_ID = "video";
+
+  private DatastoreService datastore;
 
   @Override
   public void init() throws ServletException {
-    // TODO: Implement Transcript.
+    datastore = DatastoreServiceFactory.getDatastoreService();
+  }
+
+  /**
+   * Initializes the servlet with {@code testDatastore} created during testing.
+   *
+   * <p>The unit tests need access to the datastore to check that doPost() puts
+   * the entities in datastore.
+   */
+  @VisibleForTesting
+  protected void init(DatastoreService testDatastore) {
+    datastore = testDatastore;
   }
```

Comment:
How about you call this ```initInternal(DatastoreService dataStore)``` then call it from the ```init()`` method. Having something in a main class just for tests is not good. If at all we want to do such we have it in a /testing directiory and it is usually a separate implementation of something orthogonal from the main code. 
In this case I think you are trying to find a way to initialize the servlet in tests.
```
@VisibleForTesting void initInternal(DatastoreService datastore) {
    datastore = testDatastore;
  }
```
should be enough. I don't think we need this protected here?

Reply:
Got it! I removed protected and called the test method in init(). I kept the name the same if that's okay because I felt overloading made more sense.

Reply:
This isn't necessary.  .getDatastoreService in the normal init method gets the test service if it has been .setUp.  You can .getDatastoreService in multiple places and get a reference to the same one.